gps_rviz/src/gps_handler.cpp

```diff
@@ -0,0 +1,122 @@
+#include "geometry_msgs/Vector3.h"
+#include "math.h"
+#include "visualization_msgs/Marker.h"
+#include "visualization_msgs/MarkerArray.h"
+#include "gps_handler.h"
+
+#define PI 3.14159265
+
+using namespace std;
+
+namespace gps{
+
+ geometry_msgs::Vector3 InitializingPoint, InitializingPoint_missionframe;
+ bool REFERENCE_FRAME_IS_INITIALIZED;
+ float alpha_init, beta_init;
+ 
+ visualization_msgs::MarkerArray GPS_TRACK;
+ visualization_msgs::Marker gps_points, gps_line_strip, gps_line_list;
+
+// ---------------------------------------------
+// Rotational Matrix R_0G=R_01*R_1G; R_01=Rz(a); R_1G=Ry(b)
+geometry_msgs::Vector3 transformation(geometry_msgs::Vector3 Input, float a, float b){
+	geometry_msgs::Vector3 Output;
+ 	Output.x = cos(a)*cos(b)*Input.x - sin(a)*Input.y +cos(a)*sin(b)*Input.z;
+ 	Output.y = sin(a)*cos(b)*Input.x + cos(a)*Input.y +sin(a)*sin(b)*Input.z;
+ 	Output.z = -sin(b)*Input.x + cos(b)*Input.z;
+
+return Output;
+}
+
+// -------------------------------------------------
+// Rotational Matrix plus shifting
+geometry_msgs::Vector3 transformToMissionFrame(geometry_msgs::Vector3 InputPoint){
+
+	geometry_msgs::Vector3 transformed_point;
+
+	transformed_point = transformation(InputPoint, alpha_init, beta_init); //Rotate by alpha, beta
+	transformed_point.x = transformed_point.x - InitializingPoint_missionframe.x; // shift to normalize
+	transformed_point.y = transformed_point.y - InitializingPoint_missionframe.y;
+	transformed_point.z = transformed_point.z - InitializingPoint_missionframe.z;
+
+return transformed_point;
+} 
+
+// ---------------------------------------------------
+// NavSatFix coordinates to WGS84 (cartesian) coordinates
+geometry_msgs::Vector3 gps2cart(const geometry_msgs::Vector3& gpsOut){
+	
+	float e, R, height, phi_deg, phi_rad, lambda_deg, lambda_rad; 
+
+	R = (majorAxis+minorAxis)/2; //in meters
```

Comment:
- Use unit in variable name and remove comment.
- Longer variable names.
- Code should be self-explanatory.
- Take care of double precision when using ``*`` and ``/``.
```
r_meter = (major_axis + minor_axis) / 2.0;
```